### PR TITLE
Feature - allow using buffer as input of sqip

### PR DESCRIPTION
**Pull request by gabrielperales**

### CODE REVIEW

File: packages/sqip/src/sqip.js [hunk 1]
```diff
@@ -110,17 +226,17 @@ async function processImage({ filePath, config }) {
       sqipConfig: config,
       pluginOptions,
       metadata,
-      filePath
+      // filePath
```
Comment (gabrielperales) [inline packages/sqip/src/sqip.js]: It seems to work with this line commented out 🤷‍♂. I have to check why it was needed (maybe for the CLI?)

Comment (axe312ger) [reply packages/sqip/src/sqip.js]: Some plugins require the file to exist on the file system.

We should change these plugins to create a temporary file if they do not support buffers as input

**Comment by axe312ger** [comment]
Your restructuring was necessary, thanks for that. Would be amazing if you could change the plugins that rely on `filePath` so we can get this into SQIP and release the beta :)

I could do it as well, but can't promise to find any time for that within the close future

**Comment by gabrielperales** [comment]
@axe312ger I'll try to find some time to do it :)

**Comment by axe312ger** [comment]
Hey @gabrielperales, 

I continued your PR via  #113 and added some more stuff :)

**Pull request closed without merging**